plugins/module_utils/job.py

```diff
@@ -76,9 +76,47 @@ def _get_job_output(job_id="*", owner="*", job_name="*", dd_name=""):
                 str(rc), str(err)
             )
         )
-    if not out:
-        raise RuntimeError("Failed to retrieve job output. No job output found.")
-    jobs = _parse_jobs(out)
+    jobs = []
+    if out:
+        jobs = _parse_jobs(out)
+    if not jobs:
+        jobs = _job_not_found(job_id, owner, job_name, dd_name)
+
+    return jobs
+
+
+def _job_not_found(job_id, owner, job_name, dd_name):
+    jobs = []
+
+    job = {}
+
+    job["job_id"] = job_id
+    job["job_name"] = job_name
+    job["subsystem"] = "NOTFOUND"
+    job["system"] = "NOTFOUND"
```

Comment:
Using I don't mind the use of `NOTFOUND` but we need to normalize/standardize on either use of spaces, case and verbosity. 

Let me further explain, some of the other properties have more verbose messages like `Job not found` where others are just `NOTFOUND`, `NOTFND`, so this is the verbosity difference I see and think case and use of spaces is pretty self explanatory.

Reply:
This works for me, as we discussed this probably is better than 'none' to reduce some of the ambiguity of none.

Reply:
system, subsystem, owner are now all none.  Step name within not found job is now blank.